gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -166,8 +166,9 @@ namespace gtsam {
   DiscreteKeys DecisionTreeFactor::discreteKeys() const {
     DiscreteKeys result;
     for (auto&& key : keys()) {
-        result.emplace_back(key, cardinality(key));
-      }
+      DiscreteKey discreteKey(key, cardinality(key));
+      result.push_back(discreteKey);
```

Comment:
Why?

Reply:
Ah this is leftover from a previous change that Fan pointed out. Will revert.